httputil/client.go

```diff
@@ -41,7 +101,10 @@ func NewHTTPClient(opts *ClientOptions) *http.Client {
 	}
 
 	transport := newDefaultTransport()
+	// remember the original ClientOptions.TLSConfig when making
+	// tls connection.
 	transport.TLSClientConfig = opts.TLSConfig
```

Comment:
the doc seem to say that if DialTLS is set this will be ignored (https://golang.org/pkg/net/http/#Transport) so maybe is clearer to leave this out now

Reply:
I added a comment why we need it, we use it in one of our tests to see if the config is set correctly.